### PR TITLE
README: Elaborate on what we aim to achieve with CLI testing

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
I've added a bit more detail to the README to capture what we discussed in https://github.com/sigstore/cosign/issues/2434 in regards to what we aim to achieve with CLI testing as opposed a vector based approach.

CC: @znewman01

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,29 @@ sigstore-conformance
...
+- What files are written to the disk by the client in what situations?
+- Does the client fail when given a signing certificate that isn't signed by
+  the Fulcio root CA during the signing workflow?
+- Does the client fail when given an invalid inclusion proof from Rekor during
+  the verification workflow?
+- Does the client fail when given an invalid signed certificate timestamp as
+  part of the Fulcio response in the signing workflow?
+- etc
+
+Over time, `sigstore-conformance` aims to become an informal specification that
```
Comment (znewman01) [inline README.md:22]: How does this related to the "formal" specification for Sigstore clients?

https://github.com/sigstore/architecture-docs

File: README.md [hunk 2]
```diff
@@ -8,6 +8,29 @@ sigstore-conformance
 
 `sigstore-conformance` is a conformance testing suite for Sigstore clients.
 
+This suite provides a high-level view of client behaviour as a whole and sets
+out to answer questions such as:
+- What files are written to the disk by the client in what situations?
```
Comment (znewman01) [inline README.md:13]: Hm....I don't actually view "writing files to disk" as part of the remit of most Sigstore clients. This seems to conflate "Sigstore client" with "Sigstore CLI app," and a quite specific one at that.

I actually *don't* expect that we would have a proliferation of Sigstore CLI apps. I'd expect there to be really only one (basically, an improved version of Cosign without the OCI baggage). Then, most clients of Sigstore would be application-specific (in the Python case, built into Pip, eventually).

(The remaining points all seem reasonable.)

**Review by znewman01 — APPROVED** [review]
Hey, thanks so much for writing this up! Very helpful.

Couple clarifying questions but LGTM overall

File: README.md (hunk 1 above)
Comment (tetsuo-cpp) [reply README.md:22]: I wasn't aware of that.

Maybe I can try something like:

> An official Sigstore client specification is being worked on at the moment as part of the [Sigstore Architecture Documentation](https://github.com/sigstore/architecture-docs). Once it's complete, `sigstore-conformance` aims to be able to test a client's adherence to the specification.

Comment (znewman01) [reply README.md:22]: SGTM

**Pull request merged**